### PR TITLE
Update examples to use micro URIs

### PR DESCRIPTION
**Pull request by gregmedd**
Removed dependence on the LongUriSerializer and replaced test URIs with micro-compatible URIs.

~~Depends on changes from #5.~~
~~Use <a href="https://github.com/eclipse-uprotocol/up-zenoh-example-cpp/pull/8/commits/0eec45e323b931977399385f188d4f81b34fc707">this link</a> to review changes for this PR~~

~~Use [this link](https://github.com/eclipse-uprotocol/up-zenoh-example-cpp/pull/8/files/ef150c8872b1980b5251399318ae98fee44d51ce..9663fc4f97179101bfabcc7036711302096ebcff) to review changes for this PR~~

#9 has been merged and this PR has been rebased and can be reviewed as normal.

### CODE REVIEW

**Comment by gregmedd** [comment]
Depends on changes from #5.
<h1>Use <a href="https://github.com/eclipse-uprotocol/up-zenoh-example-cpp/pull/8/commits/0eec45e323b931977399385f188d4f81b34fc707">this link</a> to review changes for this PR</h1>

File: pubsub/src/common.h
Comment (MelamudMichael) [inline pubsub/src/common.h:1]: Please move in to include folder

File: rpc/src/common.h
Comment (MelamudMichael) [inline rpc/src/common.h:1]: Please move it to include folder

File: pubsub/src/main_sub.cpp [hunk 1]
```diff
@@ -49,28 +48,28 @@ void signalHandler(int signal) {
...
     public:
-        /* in this example the same onReceive callback implementation is used to receive
-         * the three different messages , each message is differntiated by the URI
-         * it is possible to provide a different onReceive callback for each topic */
-        UStatus onReceive(const UUri& uri,
-                          const UPayload& payload,
-                          const UAttributes& attributes) const override {
-                                
-            if (TIME_URI_STRING == LongUriSerializer::serialize(uri)) {
+
```
Comment (MelamudMichael) [inline pubsub/src/main_sub.cpp:51]: can you please return the comment?

Comment (gregmedd) [reply pubsub/src/main_sub.cpp:51]: The comment was removed in one of the commits from #5 and not this PR. See above comment with link to diff view for only this PR.

File: pubsub/src/common.h
Comment (gregmedd) [reply pubsub/src/common.h:1]: Moved

File: rpc/src/common.h
Comment (gregmedd) [reply rpc/src/common.h:1]: Moved

File: pubsub/include/common.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#include <up-cpp/uri/builder/BuildUResource.h>
+
+uprotocol::v1::UUri getUri(
+        std::string_view const entName, int const entId, int const entMajorVers,
+        std::string_view const resName, int const resId) {
+
+    using namespace uprotocol::uri;
+
+    return BuildUUri()
+        .setAutority(BuildUAuthority().build())
```
Comment (billpittman) [inline pubsub/include/common.h:39]: I might need some history here.  Is this a known misspelled interface that we accept or is there another PR to address the misspelling of setAuthority.  I looked at the code, and the interface exists and error messages within the function spell Authority (with the h).  So I'm a little confused.

File: pubsub/include/common.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                .build())
+        .setResource(BuildUResource()
+                .setName(static_cast<std::string>(resName))
+                .setID(resId)
+                .build())
+        .build();
+}
+
+constexpr std::string_view ENTITY_NAME{"test.app"};
+constexpr int ENTITY_ID{2};
```
Comment (billpittman) [inline pubsub/include/common.h:53]: I might have missed something wrt ENTITY_ID in some of our chats, but what is it supposed to do, and why is it 2 ?

**Review by billpittman — APPROVED** [review]
Answer the above questions and I'm good with the PR.  Marking as approved since I don't think there's anything wrong with the code necessarily (except for the misspelling, but I think that is more me than the code).

File: pubsub/include/common.h (hunk 3 above)
Comment (evshary) [reply pubsub/include/common.h:53]: Let me create an issue for discussion #11

File: rpc/src/main_rpc_server.cpp [hunk 4]
```diff
@@ -46,49 +49,53 @@ void signalHandler(int signal) {
...
+         UStatus onReceive(UMessage &message) const override {
             /* Construct response payload with the current time */
             auto currentTime = std::chrono::system_clock::now();
             auto duration = currentTime.time_since_epoch();
             uint64_t currentTimeMilli = std::chrono::duration_cast<std::chrono::milliseconds>(duration).count();
 
             UPayload responsePayload(reinterpret_cast<const uint8_t*>(&currentTimeMilli), sizeof(currentTimeMilli), UPayloadType::VALUE);
-
+           
+            auto builder = UAttributesBuilder::response(message.attributes().source(), message.attributes().sink(), UPriority::UPRIORITY_CS0, message.attributes().id());
```
Comment (MelamudMichael) [inline rpc/src/main_rpc_server.cpp:61]: this is a bug, need to use the same UUID for response

File: pubsub/src/main_sub.cpp (hunk 1 above)
Comment (gregmedd) [reply pubsub/src/main_sub.cpp]: @stevenhartley - Can we resolve this conversation?

File: pubsub/include/common.h (hunk 2 above)
Comment (gregmedd) [reply pubsub/include/common.h:39]: Yes, unfortunately "Authority" is misspelled "Autority" in several places. We have an open issue to find and fix that: https://github.com/eclipse-uprotocol/up-cpp/issues/89

File: pubsub/include/common.h (hunk 3 above)
Comment (gregmedd) [reply pubsub/include/common.h:53]: @billpittman - Entity ID is a part of the URI that is a numeric way to locate a uEntity in the system. The broad strokes are covered in the [UUri spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc). In the case of these examples, these values are basically arbitrary. Real systems will use some mechanism to perform a lookup to get these IDs in the future (I don't recall the details off the top of my head).

@evshary - I agree that we should align the examples. Is it reasonable to move forward with this PR and follow up after we've discussed in #11?

File: rpc/src/main_rpc_server.cpp (hunk 4 above)
Comment (gregmedd) [reply rpc/src/main_rpc_server.cpp]: I don't see that change in this review. It is possible that it was an artifact of this being based on other PRs, which would have been visible if this was reviewed prior to today's rebase without using the links provided in the description above.

**Review by debruce — APPROVED** [review]
Looks good to me.

**Review by stevenhartley — APPROVED** [review]

**Pull request merged**